File: bug-ray/Domain/Project.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Project
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime Estimate { get; set; }
        public bool IsFavourate { get; set; }
        public bool IsCancelled { get; set; }

        public ICollection<ProjectContributor> Contributors { get; set; } = new List<ProjectContributor>();
        public ICollection<Discussions> Discussions { get; set; } = new List<Discussions>();


    }
}

File: bug-ray/client-app-react/src/app/models/project.ts
import { Profile } from "./profile";

export interface Project {
  id: string;
  name: string;
  description: string;
  estimate: Date | null;
  isFavourate: boolean;
  hostUsername: string;
  isCancelled: boolean;
  isContributing: boolean;
  isOwner: boolean;
  owner?: Profile;
  contributors: Profile[];
}

export class Project implements Project {
  constructor(init?: ProjectFormValues) {
    //Mapping init object with Project object
    Object.assign(this, init);
  }
}

export class ProjectFormValues {
  id?: string = undefined;
  name: string = "";
  description: string = "";
  estimate: Date | null = null;
  isFavourate: boolean = false;

  constructor(project?: ProjectFormValues) {
    if (project) {
      this.id = project.id;
      this.name = project.name;
      this.description = project.description;
      this.estimate = project.estimate;
      this.isFavourate = project.isFavourate;
    }
  }
}
